dev/com.ibm.ws.microprofile.openapi/src/com/ibm/ws/microprofile/openapi/impl/validation/OAuthFlowsValidator.java

```diff
@@ -34,6 +37,45 @@ private OAuthFlowsValidator() {}
     /** {@inheritDoc} */
     @Override
     public void validate(ValidationHelper helper, Context context, String key, OAuthFlows t) {
-        // TODO Auto-generated method stub
+        if (t.getImplicit() != null) {
+            OAuthFlow implicit = t.getImplicit();
+            String type = "implicit OAuthFlow";
+            if (StringUtils.isNotBlank(implicit.getTokenUrl())) {
+                final String message = Tr.formatMessage(tc, "nonApplicableField", "tokenUrl", implicit.getTokenUrl(), type);
+                helper.addValidationEvent(new ValidationEvent(ValidationEvent.Severity.WARNING, null, message));
+            } else {
+                ValidatorUtils.validateRequiredField(implicit.getAuthorizationUrl(), context, "authorizationUrl").ifPresent(helper::addValidationEvent);
```

Comment:
This check should not be in an else-block. The validator should gather all events from validating its fields (not one or the other).

Same is true for the checks below.